test/graaflib/graph_test.cpp

```diff
@@ -85,5 +116,91 @@ namespace graaf {
         ASSERT_TRUE(graph.has_vertex(vertex_id_1));
         ASSERT_TRUE(graph.has_vertex(vertex_id_2));
     }
+    TYPED_TEST(GraphTest, AddEdge)
+    {
+        using graph_t = typename TestFixture::graph_t;
+
+        graph_t graph{};
+        const auto vertex_id_1{graph.add_vertex(10)};
+        const auto vertex_id_2 = graph.vertex_count() + 1;
+
+        ASSERT_THROW({
+            try
+            {
+                // Call the add_edge function with a non-existing vertex ID
+                graph.add_edge(vertex_id_1, vertex_id_2, 100);
+                // If the above line doesn't throw an exception, fail the test
+                FAIL() << "Expected std::out_of_range exception, but no exception was thrown.";
```

Comment:
[minor]:
Properly checking this in gtest is quite verbose 😬 

However, if you want you could save a line by removing the `FAIL` call, as the outer most `ASSERT_THROW` would fail the test in case the `add_edge` call does not throw. Same for the other two tests below.

```suggestion
                // Call the add_edge function with a non-existing vertex ID
                graph.add_edge(vertex_id_1, vertex_id_2, 100);
```